[wr1159] iP

wr1159 opened the pull request:
## **Sisyphus** 🪨 

> "One must imagine Sisyphus happy." - Albert Camus [[Read More]](https://www.peabodylibrary.org/freeforall/?p=6998)

### **Sisyphus** helps you manage the following:
* To-Dos
* Deadlines
* Events

### Sisyphus has the following features:

1. ~Interesting~ Very Interesting Personality
2. CLI support for **_faster_** input and output
3. Adding, Deleting, Marking and Reading of Tasks and more!

### Roadmap

- [x] To-Do
- [x] Deadlines
- [x] Events
- [ ] Assigning **priority** to tasks
- [ ] Export data as CSV

### Code
Here is the `main` function of `Sisyphus.java` which runs the program:
```java
    public static void main(String[] args) {
        Sisyphus sisyphus = new Sisyphus();
        sisyphus.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wr1159 committed:
Rename bot and add basic greetings

wr1159 committed:
Add echo function and reformatted greeting to include logo

wr1159 committed:
Add addAndList function to meet Level-2 requirements

wr1159 committed:
Add mark and unmark functionalities to meet Level 3 requirements

wr1159 committed:
Add ToDo, Deadline, Event and support of the respective commands to meet Level 4 requirements

wr1159 committed:
Clean up formatting

wr1159 committed:
Add text-ui test and update string for unmark

wr1159 committed:
Add exception handling to meet level 5 requirements

wr1159 committed:
Add delete function

wr1159 committed:
Remove redundant -1 value in Mark

wr1159 committed:
Add TaskList object

wr1159 committed:
Add optional isDone param to constructor to task, deadline, event, todo

wr1159 committed:
Add toSaveFormat function for Task and children of Task

wr1159 committed:
Fix saveFormat to correspond to event type

wr1159 committed:
Add Storage class and integrated Storage into chatbot to save TaskList into local storage

wr1159 committed:
Merge branch 'branch-level-7' to add Storage Class

wr1159 committed:
Use LocalDate for deadline

wr1159 committed:
Change deadline date save format to yyyy-mm-dd

wr1159 committed:
Merge branch 'branch-level-8' to include LocalDate deadline date

wr1159 committed:
Add getLastTask function in TaskList and add Ui class to handle reading and printing of Strings

wr1159 committed:
Rename Duke to Sisyphus while including parser, ui classes

wr1159 committed:
Merge branch 'A-MoreOOP'

wr1159 committed:
Remove unused code in Sisyphus

wr1159 committed:
Add sisyphus and sisyphus.task packages

wr1159 committed:
Merge branch 'A-Packages'

wr1159 committed:
Merge 'add-gradle-support' to add Gradle warpper files

wr1159 committed:
Change gradle build to use sisyphus instead of duke

wr1159 committed:
Fix delete index error

wr1159 committed:
Add tests for Deadline and TaskList

wr1159 committed:
Merge branch 'A-JUnit' to test Deadline and TaskList

wr1159 committed:
Fix Typo in printAddTodo

wr1159 committed:
Add JavaDocs comments

wr1159 committed:
Merge branch 'A-JavaDoc' for JavaDoc

wr1159 committed:
Rename function name to follow coding standard

wr1159 committed:
Resolve merge conflicts

wr1159 committed:
Refactor to use getSize instead of size

wr1159 committed:
Add hasKeyword in Task and findMatchingTasks in TaskList

wr1159 committed:
Integrate find feature with findMatchingTasks

wr1159 committed:
Merge branch 'Level-9'

wr1159 committed:
Setup checkstyle

wr1159 committed:
Fix styling errors

wr1159 committed:
Merge branch 'A-CheckStyle'

ryamgoh commented on src/main/java/sisyphus/Parser.java:
```diff
@@ -0,0 +1,165 @@
+package sisyphus;
```
Well done on the overall code! Looks very clean

Though I think you can try to separate into more packages for example:
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/122634188/d1f0a1bc-1d93-433a-97b5-f10af9fa0e6d)

This will make the code more segmented and easier to follow in a large code base! :)

ryamgoh commented on src/main/java/sisyphus/Storage.java:
```diff
@@ -0,0 +1,114 @@
+ * Storage class that stores and reads a local data file in the user's computer.
+ */
+public class Storage {
+    private String currentDirectory = System.getProperty("user.dir");
```
I think it will be good to give some javadocs for your fields (even though its private haha)

ryamgoh commented on src/main/java/sisyphus/Parser.java:
```diff
@@ -0,0 +1,165 @@
+            ui.printUnmarkTask(taskList, index);
+            break;
+        }
+        case ("delete"): {
```
nice getting rid of if else statements and using switch case statements

ryamgoh commented on src/main/java/sisyphus/Parser.java:
```diff
@@ -0,0 +1,165 @@
+                index = Integer.parseInt(params.split(" ")[0]) - 1;
+                taskList.markTask(index);
+                storage.writeFile(taskList);
+            } catch (Exception e) {
```
Would be good to remove the pokemon catching

ryamgoh commented on src/main/java/sisyphus/Ui.java:
```diff
@@ -0,0 +1,178 @@
+
+    private Scanner scanner;
+
+    public Ui() {
```
can put javadocs here even though its a constructor (sometimes u also have overlaoded constructors)

wjacobw commented on src/main/java/sisyphus/Ui.java:
```diff
@@ -0,0 +1,178 @@
+ */
+public class Ui {
+
+    private static final String NAME = "sisyphus.Sisyphus";
```
Nice that you declared these constants at the class level using the final keyword and in uppercase letters, following naming conventions.

wjacobw commented on src/test/java/sisyphus/task/TaskListTest.java:
```diff
@@ -0,0 +1,87 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TaskListTest {
```
You've named the test class appropriately with the suffix "Test," which is a good practice. It's clear that this class is intended for testing the TaskList class.

wjacobw commented on src/main/java/sisyphus/Parser.java:
```diff
@@ -142,9 +142,14 @@ public void runCommand(String fullCommand, TaskList taskList, Storage storage, U
             ui.printAddEvent(taskList);
             break;
         }
+        case ("find"): {
```
Using switch statement is a good way to structure command-based operations.

wjacobw commented on src/main/java/Duke.java:
```diff
@@ -53,7 +53,7 @@ public static void handleCommand() throws SisyphusException {
         Scanner scanner= new Scanner(System.in);
         String input, command, params = "";
```
Maybe it's good practice to declare and initialize variables on separate lines? I'm not sure haha

wjacobw reviewed (commented):
LGTM!

wr1159 committed:
Change name to Sisyphus

wr1159 committed:
Add packages and add javaDocs comments

wr1159 committed:
Add javadocs comments for private field

wr1159 committed:
Merge branch 'Coding-Standards'

wr1159 committed:
Add javafx into build gradle

wr1159 committed:
Change order of import to meet checkstyle

wr1159 committed:
Add basic hello world scene in javafx

wr1159 committed:
Add input and send button into scene

wr1159 committed:
Increase size of anchor pane

wr1159 committed:
Add send message functionality

wr1159 committed:
Add imageview into dialog

wr1159 committed:
Add flip function

wr1159 committed:
Integrate response into GUI

wr1159 committed:
Add FXML, rectangle to chat

wr1159 committed:
Update printed text

wr1159 committed:
Fix CheckStyle

wr1159 committed:
Merge branch 'branch-Level-10'

wr1159 committed:
Add assertions to code

This commit introduces assertions to the codebase by enabling it in
`gradle.build`.

There are several methods in the code that should hold at various points
to ensure the safety of our program. This includes certain
uninstantialised objects which could be null values.

By using assertions, we are able to put assertion points that ensure
our variables are not null values.

wr1159 committed:
Merge pull request #2 from wr1159/branch-A-Assertions

Add assertions to code

wr1159 committed:
Refactor loadData function to smaller functions

Created functions such as `addEventFromStorage`, `addToDoFromStorage`,
`addDeadlineFromStorage` to be used in loadData.

This will reduce the length of the function and helps with SLAP,
increasing code quality.

wr1159 committed:
Refactor runCommand to smaller commands

This commit adds sveral new commands for each case like
`runDeadlineCommand`, `runEventCommand`, `runMarkCommand` and etc.

This is done by moving the contents of each case into another function.

This will now help with SLAP and lessen the code length of the function.

wr1159 committed:
Add javaDocs coments for new command functions

wr1159 committed:
Change header comments to fit code standard

wr1159 committed:
Fix delete command to print correct output string

wr1159 committed:
Merge pull request #3 from wr1159/branch-A-CodeQuality

Refactor to improve Code Quality

wr1159 committed:
Add comment to indicate C-BetterSearch

wr1159 committed:
Merge pull request #4 from wr1159/branch-BCD-Extension

Indicate C-BetterSearch as part of BCD Extension

wr1159 committed:
Fix length of runDeadlineCommand

wr1159 committed:
Add Ui.png to docs folder

wr1159 committed:
Add basic user guide with features and description

wr1159 committed:
Merge branch 'branch-A-UserGuide'

wr1159 committed:
Abstract DateFormatter into its own class and function

wr1159 committed:
Add LocalDate support for Event

wr1159 committed:
Update user guide to include LocalDate support for event

wr1159 committed:
Add slate theme for documentation

wr1159 committed:
Merge branch 'Fix-Event-Dates'

wr1159 committed:
Fix indentation and punctuation in user guide

wr1159 committed:
Refactor to prepare for release

wr1159 committed:
Fix Checkstyle

wr1159 committed:
Switch themes to cayman

wr1159 committed:
Revert theme back to slate

wr1159 committed:
Remove theme for user guide

wr1159 committed:
Add slate theme

wr1159 committed:
Change example event commands in UG

wr1159 committed:
Update Ui.png

wr1159 committed:
Merge branch 'Fix-Event-UG'

wr1159 committed:
Update build.gradle to pass smoke test
